src/DelayBucket.cc

```diff
@@ -46,8 +56,14 @@ DelayBucket::bytesIn(int qty)
 void
 DelayBucket::init(DelaySpec const &rate)
 {
-    level() = (int) (((double)rate.max_bytes *
-                      Config.Delay.initial) / 100);
+    SetToNaturalSumOrMax(level_, rate.max_bytes, Config.Delay.initial);
+    // getting around possible integer overflows without turning to floats
+    if (level_ < MaxValue(level_)) {
+        level_ /= 100;
+    } else {
+        // Config.Delay.initial is always <= 100
```

Comment:
Config.Delay.initial (delay_initial_bucket_level) is documented as '(percent, 0-100)', but unfortunately the parsed value is not validated against these limits. Since we rely on the 0-100 range here, we need to enforce this requirement, adjusting the parser (to be consistent with the documentation) and probably adding an assertion here.
BTW, a similar client_delay_initial_bucket_level is documented as (percent, 0-no_limit) - i.e. we have some percentage parameters without limitation.
I wonder whether Squid checks values for other 'percent' parameters. I did not look through them all, but for example, 'digest_rebuild_chunk_percentage' is not validated either.